streams/src/main/java/org/apache/kafka/streams/state/internals/TimestampedKeyAndJoinSide.java

```diff
@@ -33,28 +34,36 @@
 public class TimestampedKeyAndJoinSide<K> {
     private final K key;
     private final long timestamp;
-    private final boolean leftSide;
+    private final JoinSide joinSide;
 
-    private TimestampedKeyAndJoinSide(final boolean leftSide, final K key, final long timestamp) {
+    private TimestampedKeyAndJoinSide(final JoinSide joinSide, final K key, final long timestamp) {
         this.key = Objects.requireNonNull(key, "key cannot be null");
-        this.leftSide = leftSide;
+        this.joinSide = joinSide;
         this.timestamp = timestamp;
     }
 
     /**
-     * Create a new {@link TimestampedKeyAndJoinSide} instance if the provide {@code key} is not {@code null}.
+     * Create a new {@link TimestampedKeyAndJoinSide} instance if the provided {@code key} is not {@code null}.
      *
-     * @param leftSide True if the key is part of the left join side; False if it is from the right join side
+     * @param joinSide Whether the key is part of the {@link JoinSide#LEFT} side; or it is from the {@link JoinSide#RIGHT} side
      * @param key      the key
      * @param <K>      the type of the key
-     * @return a new {@link TimestampedKeyAndJoinSide} instance if the provide {@code key} is not {@code null}
+     * @return a new {@link TimestampedKeyAndJoinSide} instance if the provided {@code key} is not {@code null}
      */
-    public static <K> TimestampedKeyAndJoinSide<K> make(final boolean leftSide, final K key, final long timestamp) {
-        return new TimestampedKeyAndJoinSide<>(leftSide, key, timestamp);
+    public static <K> TimestampedKeyAndJoinSide<K> make(final JoinSide joinSide, final K key, final long timestamp) {
```

Comment:
Since this is only used in tests now, I think you can remove this and replace the test call-sites with the new functions.

Make sure to copy the javadoc to the new signatures too.